api/types/types.go

```diff
@@ -190,3 +190,42 @@ type Workflow struct {
 	Parameters    []WorkflowParam `yaml:"parameters"`
 	Source        string          `valid:"isValidWorkflow" yaml:"source"`
 }
+
+// Node describes a node
+type Node struct {
+	UUID     string
+	Type     string `valid:"nonzero"`
+	Name     string `valid:"nonzero"`
+	Metadata string
```

Comment:
What about making Metadata a map[string]interface{}, so we just have to call defToMetadata at creation time ?